src/CommunityToolkit.Maui.CameraView/Views/CameraManager.ios.cs

```diff
@@ -0,0 +1,273 @@
+
+    protected virtual partial void PlatformConnect()
+    {
+        if (cameraProvider.AvailableCameras.Count < 1)
+        {
+            throw new InvalidOperationException("There's no camera available on your device.");
+        }
+
+        PlatformStart();
+    }
+
+    protected virtual partial void PlatformStart()
+    {
+        if (currentCamera is null || captureSession is null)
+        {
+            return;
+        }
+
+        captureSession.BeginConfiguration();
+
+        foreach (var input in captureSession.Inputs)
+        {
+            captureSession.RemoveInput(input);
+            input.Dispose();
+        }
+
+        captureDevice = currentCamera.CaptureDevice ?? throw new NullReferenceException();
+        captureInput = new AVCaptureDeviceInput(captureDevice, out var err);
+        captureSession.AddInput(captureInput);
+
+        if (photoOutput is null)
+        {
+            photoOutput = new AVCapturePhotoOutput();
+            captureSession.AddOutput(photoOutput);
+        }
+
+        UpdateCaptureResolution(cameraView.CaptureResolution);
+
+        captureSession.CommitConfiguration();
+        captureSession.StartRunning();
+        Initialized = true;
+        Loaded?.Invoke();
+    }
+
+    protected virtual partial void PlatformStop()
+    {
+        if (captureSession is null)
+        {
+            return;
+        }
+
+        if (captureSession.Running)
+        {
+            captureSession.StopRunning();
+        }
+
+        Initialized = false;
+    }
+
+    protected virtual partial void PlatformDisconnect()
+    {
+    }
+
+    protected virtual async partial void PlatformTakePicture()
+    {
+        ArgumentNullException.ThrowIfNull(photoOutput);
+
+        var capturePhotoSettings = AVCapturePhotoSettings.FromFormat(codecSettings);
+        capturePhotoSettings.FlashMode = photoOutput.SupportedFlashModes.Contains(flashMode) ? flashMode : photoOutput.SupportedFlashModes.First();
+
+        var wrapper = new AVCapturePhotoCaptureDelegateWrapper();
+
+        photoOutput.CapturePhoto(capturePhotoSettings, wrapper);
+
+        var result = await wrapper.Task;
+        var data = result.Photo.FileDataRepresentation;
+
+        if (data is null)
+        {
+            // TODO: Pass NSError information
+            cameraView.OnMediaCapturedFailed();
+            return;
+        }
+
+        var dataBytes = new byte[data.Length];
+        Marshal.Copy(data.Bytes, dataBytes, 0, (int)data.Length);
+
+        cameraView.OnMediaCaptured(new MemoryStream(dataBytes));
+    }
+
+    public partial void UpdateFlashMode(CameraFlashMode flashMode)
+    {
+        this.flashMode = flashMode.ToPlatform();
+    }
+
+    public partial void UpdateZoom(float zoomLevel)
+    {
+        if (!Initialized || captureDevice is null)
+        {
+            return;
+        }
+
+        if (zoomLevel < (float)captureDevice.MinAvailableVideoZoomFactor || zoomLevel > (float)captureDevice.MaxAvailableVideoZoomFactor)
```

Comment:
I think this could be rewritten as:

```suggestion
        if (zoomLevel is < (float)captureDevice.MinAvailableVideoZoomFactor or  > (float)captureDevice.MaxAvailableVideoZoomFactor)
```

which might make it a tiny bit easier to read.

Reply:
It seems like the [right-hand part of a relational pattern must be a constant expression](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#relational-patterns), so this syntax is not supported here.